sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1749,6 +1749,19 @@ object SQLConf {
     .checkValue(v => v > 0, "The min partition number must be a positive integer.")
     .createOptional
 
+  val FILES_MAX_DESIRED_PARTITION_NUM = buildConf("spark.sql.files.maxDesiredPartitionNum")
```

Comment:
Maybe simply use `spark.sql.files.maxPartitionNum`? 

The existing `spark.sql.files.minPartitionNum` is not guaranteed either.

```
val FILES_MIN_PARTITION_NUM = buildConf("spark.sql.files.minPartitionNum")
    .doc("The suggested (not guaranteed) minimum number of split file partitions. " +
    ...
```

Reply:
make sense.